megamek/src/megamek/client/ratgenerator/AbstractUnitRecord.java

```diff
@@ -155,5 +155,9 @@ public static int parseUnitType(String typeName) {
 				return -1;
 		}
 	}
+	
+	public int compareTo(AbstractUnitRecord o2) {
+	    return this.getChassis().compareTo(o2.getChassis());
+	}
```

Comment:
The two classes that extend this one should sort on different fields. Only ChassisRecord uses the chassis, so there is no good reason to implement it this way in the base class.